lib/vm/src/instance/mod.rs

```diff
@@ -1351,23 +1334,6 @@ unsafe fn get_memory_slice<'instance>(
     slice::from_raw_parts_mut(memory.base, memory.current_length.try_into().unwrap())
 }
 
-fn check_memory_init_bounds(
-    instance: &Instance,
-    data_initializers: &[DataInitializer<'_>],
-) -> Result<(), Trap> {
-    for init in data_initializers {
-        let start = get_memory_init_start(init, instance);
-        unsafe {
-            let mem_slice = get_memory_slice(init, instance);
-            if mem_slice.get_mut(start..start + init.data.len()).is_none() {
-                return Err(Trap::new_from_runtime(TrapCode::HeapSetterOutOfBounds));
```

Comment:
Is `HeapSetterOutOfBounds ` being used somewhere else? If not, can we remove it from the Enum?

Reply:
This looks like it might not be used anywhere else... I'll revisit this before shipping, good call!

Reply:
Okay, so I'm going to keep this for now because while it currently is not used at all, judging by the fact that the other one (for tables) is used in `Table::copy`, I suspect that during the implementation of bulk memory we may need it again. And if so it'll be easier to keep it than remove it and readd it. If it's not needed for bulk memory, then we can actually delete it though!

Reply:
Ok, can you add a TODO over those enums mentioning that precisely? (that way will be easier to remove if we forgot)

Reply:
good idea! done!